pkg/reader/usdc_reader_test.go

```diff
@@ -0,0 +1,360 @@
+	).Return(nil, errors.New("error")).Maybe()
+
+	validSequence := []types.Sequence{
+		{
+			Data: &MessageSentEvent{Arg0: make([]byte, 128)},
+		},
+	}
+
+	validChain := cciptypes.ChainSelector(sel.ETHEREUM_MAINNET_ARBITRUM_1.Selector)
+	validReader := reader.NewMockContractReaderFacade(t)
+	validReader.EXPECT().Bind(mock.Anything, mock.Anything).Return(nil)
+	validReader.EXPECT().QueryKey(
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+		mock.Anything,
+	).Return(validSequence, nil).Maybe()
+
+	tokensConfigs := map[cciptypes.ChainSelector]pluginconfig.USDCCCTPTokenConfig{
+		faultyChain: {
+			SourcePoolAddress:            "0x000000000000000000000000000000000000000A",
+			SourceMessageTransmitterAddr: "0x000000000000000000000000000000000000000B",
+		},
+		emptyChain: {
+			SourcePoolAddress:            "0x000000000000000000000000000000000000000C",
+			SourceMessageTransmitterAddr: "0x000000000000000000000000000000000000000D",
+		},
+		validChain: {
+			SourcePoolAddress:            "0x000000000000000000000000000000000000000E",
+			SourceMessageTransmitterAddr: "0x000000000000000000000000000000000000000F",
+		},
+	}
+
+	contactReaders := map[cciptypes.ChainSelector]contractreader.ContractReaderFacade{
+		faultyChain: faultyReader,
+		emptyChain:  emptyReader,
+		validChain:  validReader,
+	}
+
+	tokens := map[exectypes.MessageTokenID]cciptypes.RampTokenAmount{
+		exectypes.NewMessageTokenID(1, 1): {
+			ExtraData: NewSourceTokenDataPayload(11, emptyChainCCTP).ToBytes(),
+		},
+	}
+
+	usdcReader, err := NewUSDCMessageReader(tokensConfigs, contactReaders)
+	require.NoError(t, err)
+
+	tt := []struct {
+		name           string
+		sourceSelector cciptypes.ChainSelector
+		destSelector   cciptypes.ChainSelector
+		expectedMsgIDs []exectypes.MessageTokenID
+		errorMessage   string
+	}{
+		{
+			name:           "should return empty dataset when chain doesn't have events",
+			sourceSelector: emptyChain,
+			destSelector:   faultyChain,
+			expectedMsgIDs: []exectypes.MessageTokenID{},
+		},
+		{
+			name:           "should return error when chain reader errors",
+			sourceSelector: faultyChain,
+			destSelector:   emptyChain,
+			errorMessage:   "error querying contract reader for chain 6433500567565415381",
+		},
+		{
+			name:           "should return error when CCTP domain is not supported",
+			sourceSelector: emptyChain,
+			destSelector:   cciptypes.ChainSelector(2),
+			errorMessage:   "destination domain not found for chain ChainSelector(2)",
+		},
+		{
+			name:           "should return error when CCTP domain is not supported",
+			sourceSelector: cciptypes.ChainSelector(sel.POLYGON_MAINNET.Selector),
+			destSelector:   emptyChain,
+			errorMessage:   "no contract bound for chain 4051577828743386545",
+		},
+		{
+			name:           "valid chain return events but nothing is matched",
+			sourceSelector: validChain,
+			destSelector:   emptyChain,
+			expectedMsgIDs: []exectypes.MessageTokenID{},
+		},
```

Comment:
Need a test from 2 valid chains and see the expected messages?

Reply:
I don't want to go that deep, getting results will require a ton of mocking and byte operations to match events. It's covered in the integration tests in the chainlink repository, I don't want to reinvent the wheel at this stage. It's a simple unit test with basic flows